api-authentication/server/server.js

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import 'dotenv/config';
+import pg from 'pg';
+import argon2 from 'argon2';
+import express from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.post('/api/auth/sign-up', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) {
+      throw new ClientError(400, 'username and password are required fields');
+    }
+    const hashedPassword = await argon2.hash(password);
+    const sql = `
+      insert into "users" ("username", "hashedPassword")
+      values ($1, $2)
+      returning *
+    `;
+    const params = [username, hashedPassword];
+    const result = await db.query(sql, params);
+    const [user] = result.rows;
+    res.status(201).json(user);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/auth/sign-in', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) {
+      throw new ClientError(401, 'invalid login');
+    }
+
+    /* your code starts here */
+
+    const sql = ` select *
+    from "users"
+    where "username" = $1
+    ;
+    `;
+    const params = [username];
+    const result = await db.query(sql, params);
+    if (result.rows === []) {
+      throw new Error(401, 'invalid username');
+    }
+    const hashedPw = result.rows[0].hashedPassword;
+    const isMatchingPw = await argon2.verify(hashedPw, password);
+    if (!isMatchingPw) throw new Error(401, 'invalid password');
+    if (isMatchingPw) {
```

Comment:
This check isn't needed. The only way you can get to this line is if `isMatchingPw` is true.